File: Personal/Collatz Conjeture/Collatz.py
import random

def Collatz(N):
    if N == 1:
        return True
    if N % 2 == 0:
        N = N/2
    else:
        N = N*3 +1
    return Collatz(N)
    
def main():
    for N in range(500,10000000000):
        result = Collatz(N)
        if result:
            print("Conjeture solved for " + str(N))
            
        else:
            print("Conjeture not solved for " + str(N))
        
if __name__ == "__main__":
    main()

File: Personal/Collatz Conjeture/Collatz.c

int collatz(int N)
{
    int M;
    for (int n = 0; n <= 10; n++){
        for(int m = 0; m <= 10; m++){
                M = ((N * (1/2)^n *3)+1)^m;
                if (M == N){
                    return 1;
                }
        }
    }
}

void main(){
    for (int i = 0; i <= 100000; i++){
        int N = random();
        if (collatz(N)){
            println("Collatz Conjeture Solved, loop created with" + str(N));
        } else {
            println("Loop not created for" + str(N));
        }
    }
    return 0;
}